packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -232,26 +232,33 @@ class CameraValue {
 /// To show the camera preview on the screen use a [CameraPreview] widget.
 class CameraController extends ValueNotifier<CameraValue> {
   /// Creates a new camera controller in an uninitialized state.
+  /// Deprecated, use [withSettings].
   CameraController(
     CameraDescription description,
-    this.resolutionPreset, {
-    this.enableAudio = true,
+    ResolutionPreset resolutionPreset, {
+    bool enableAudio = true,
+    this.imageFormatGroup,
+  })  : mediaSettings = MediaSettings(
+            resolutionPreset: resolutionPreset, enableAudio: enableAudio),
+        super(CameraValue.uninitialized(description));
+
+  /// Creates a new camera controller in an uninitialized state, using specified media settings like fps and bitrate.
+  CameraController.withSettings(
+    CameraDescription description, {
+    this.mediaSettings,
     this.imageFormatGroup,
   }) : super(CameraValue.uninitialized(description));
 
   /// The properties of the camera device controlled by this controller.
   CameraDescription get description => value.description;
 
-  /// The resolution this controller is targeting.
+  /// The media settings this controller is targeting.
   ///
-  /// This resolution preset is not guaranteed to be available on the device,
+  /// This media settings are not guaranteed to be available on the device,
   /// if unavailable a lower resolution will be used.
   ///
-  /// See also: [ResolutionPreset].
-  final ResolutionPreset resolutionPreset;
```

Comment:
Removing these will make this a breaking change. These can be replaced with getters that return the value from `MediaSettings`.